tools/bcdb_mock/mw/auth.go

```diff
@@ -0,0 +1,103 @@
+package mw
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/zos/pkg/schema"
+	"github.com/threefoldtech/zos/tools/bcdb_mock/pkg/phonebook/types"
+	"github.com/zaibon/httpsig"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type mongoKeyGetter struct {
+	db *mongo.Database
+}
+
+func (m mongoKeyGetter) GetKey(id string) interface{} {
```

Comment:
Shouldn't this be different for different endpoints. for example, the node endpoints where the node initialize the code should check against the Node public key, not the user public key

Reply:
You are correct.